File: CSE 4.1/CSE 4108 (AI Lab)/Assignments/Assignment-1/findGrandparents.pl
parent('Shakib' , 'Tamim').
parent('Tamim' , 'Shohan').
parent('Tamim' , 'Mushfiq').
parent('Riad' , 'Shakib').
parent('Riad' , 'Sabbir').
parent('Shohan', 'Afif').

grandparent(X, Z) :-
    parent(X, Y), parent(Y, Z).

findGp :-
    write(' Grandchildren: '), read(Z), write('Grandparent: '),
    grandparent(Gp,Z), write(Gp), tab(5), fail.

findGp.

File: CSE 4.1/Labs/CSE 4108 (Artificial Intelligence Lab)/Assignments/Assignment-1/findGrandparents.py
tupleList1 = [('parent', 'Shakib', 'Tamim'),
              ('parent', 'Tamim', 'Shohan'),
              ('parent', 'Tamim', 'Mushfiq'),
              ('parent', 'Riad', 'Shakib'),
              ('parent', 'Riad', 'Sabbir'),
              ('parent', 'Shohan', 'Afif')]
X = str(input("Grandchildren:"))
print('Grandparent:', end=' ')
i, j = 0, 0
for i in range(6):
    if (tupleList1[i][0] == 'parent') & (tupleList1[i][2] == X):
        for j in range(6):
            if (tupleList1[j][0] == 'parent') & (tupleList1[i][1] == tupleList1[j][2]):
                print(tupleList1[j][1], end=' ')
